src/libcore/mem.rs

```diff
@@ -900,10 +900,16 @@ pub fn discriminant<T>(v: &T) -> Discriminant<T> {
     }
 }
 
+// FIXME: Reference `MaybeUninit` from these docs, once that is stable.
 /// A wrapper to inhibit compiler from automatically calling `T`’s destructor.
 ///
 /// This wrapper is 0-cost.
 ///
+/// `ManuallyDrop<T>` is subject to the same layout optimizations as `T`.
+/// As a consequence, it has *no effect* on the assumptions that the compiler makes
+/// about all values being initialized at their type.  In particular, initializing
+/// a `ManuallyDrop<&T>` with [`mem::zeroed`] is undefined behavior.
```

Comment:
It is very easy to miss `&` near that `T`. I spent a few minutes being confused. Can we give a better, more-in-your-face, example here? Like `ManuallyDrop<NonNull<T>>`?

Reply:
Maybe `ManuallyDrop<&mut T>`?

Reply:
I like `&mut T`.